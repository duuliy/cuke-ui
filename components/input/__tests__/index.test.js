import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Input from "../index";
import { SuccessIcon, CloseCircleIcon } from "../../icon";

describe("<Input/>", () => {
  it("should render Input", () => {
    const wrapper = render(
      <div>
        <Input type="text" />
        <Input type="password" placeholder="请输入" />
        <Input type="number" placeholder="请输入" />
        <Input readonly value="我是只读" />
        <Input disabled placeholder="禁用" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render addon with input", () => {
    const wrapper = render(
      <div>
        <Input addonBefore={"www"} placeholder="请输入" />
        <Input addonAfter={".com"} placeholder="填写网址" />
        <Input
          addonBefore={"https://"}
          addonAfter={".cn"}
          placeholder="www.lijinke"
        />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should render prefix and suffix with input", () => {
    const wrapper = render(
      <div>
        <Input prefix={<SuccessIcon />} placeholder="请输入" />
        <Input
          suffix={<SuccessIcon />}
          placeholder="请输入"
          style={{ margin: "10px 0" }}
        />
        <Input
          prefix={<SuccessIcon />}
          suffix={<SuccessIcon />}
          placeholder="请输入"
        />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(".cuke-input-prefix")).toHaveLength(2);
    expect(wrapper.find(".cuke-input-suffix")).toHaveLength(2);
    expect(wrapper.find(".cuke-input-has-prefix")).toHaveLength(2);
    expect(wrapper.find(".cuke-input-has-suffix")).toHaveLength(2);
  });

  it("should render CloseCircleIcon", () => {
    const wrapper = shallow(
      <Input allowClear value={1} placeholder="请输入" />
    );

    expect(wrapper.find(CloseCircleIcon)).toHaveLength(1);
  });

  it("should emit onChange events", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Input onChange={onChange} type="text" />);

    wrapper.find(".cuke-input").simulate("change", {
      persist: () => {},
      target: {
        value: null
      }
    });
    expect(onChange).toHaveBeenCalled();
  });

  it("should emit onChange events", () => {
    const onClear = jest.fn();
    const wrapper = shallow(
      <Input onClear={onClear} allowClear value={1} suffix={1} type="text" />
    );

    wrapper.find(".cuke-input-clear").simulate("click");
    expect(onClear).toHaveBeenCalled();
  });

  it("should render custom size", () => {
    const wrapper = render(
      <div>
        <Input placeholder="small" size="small" />
        <Input placeholder="default" />
        <Input placeholder="large" size="large" />
      </div>
    );
    expect(wrapper.find(".cuke-input-small").length === 1);
    expect(wrapper.find(".cuke-input-large").length === 1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should set new state when props value update", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Input onChange={onChange} type="text" value={1} />
    );

    expect(wrapper.state().value).toBe(1);
    wrapper.setProps({
      value: 3
    });
    expect(wrapper.state().value).toBe(3);
  });

  it("should allow clear and render close icon when has value", () => {
    const wrapper = shallow(<Input allowClear placeholder="请输入" />);
    expect(wrapper.find(".cuke-input-clear")).toHaveLength(0);
    wrapper.setProps({
      value: 111
    });
    expect(wrapper.find(".cuke-input-clear")).toHaveLength(1);
    wrapper.setProps({
      allowClear: false,
      value: undefined
    });
    expect(wrapper.find(".cuke-input-clear")).toHaveLength(0);
  });

  it("should clear value when clear button clicked", () => {
    const wrapper = shallow(
      <Input allowClear placeholder="请输入" value={333} />
    );
    expect(wrapper.find(".cuke-input-clear")).toHaveLength(1);
    wrapper.setProps({ value: 111 });
    wrapper.find(".cuke-input-clear").simulate("click");
    setTimeout(() => {
      expect(wrapper.state().value).toBe("");
    });
  });

  it("should render input password", () => {
    const wrapper = render(<Input.Password placeholder="请输入" value={333} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(".cuke-input-password")).toHaveLength(1);
  });
});
