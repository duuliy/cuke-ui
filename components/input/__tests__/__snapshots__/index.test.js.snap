// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render Input 1`] = `
<div>
  <input
    class="cuke-input"
    placeholder=""
    type="text"
    value=""
  />
  <input
    class="cuke-input"
    placeholder="请输入"
    type="password"
    value=""
  />
  <input
    class="cuke-input"
    placeholder="请输入"
    type="number"
    value=""
  />
  <input
    class="cuke-input"
    placeholder=""
    readonly=""
    type="text"
    value="我是只读"
  />
  <input
    class="cuke-input cuke-input-disabled"
    disabled=""
    placeholder="禁用"
    type="text"
    value=""
  />
</div>
`;

exports[`<Input/> should render addon with input 1`] = `
<div>
  <span
    class="cuke-input-group cuke-input-group-addon-before"
  >
    <span
      class="cuke-input-group-addon"
    >
      www
    </span>
    <input
      class="cuke-input"
      placeholder="请输入"
      type="text"
      value=""
    />
  </span>
  <span
    class="cuke-input-group cuke-input-group-addon-after"
  >
    <input
      class="cuke-input"
      placeholder="填写网址"
      type="text"
      value=""
    />
    <span
      class="cuke-input-group-addon"
    >
      .com
    </span>
  </span>
  <span
    class="cuke-input-group cuke-input-group-addon-before cuke-input-group-addon-after cuke-input-group-addon-all"
  >
    <span
      class="cuke-input-group-addon"
    >
      https://
    </span>
    <input
      class="cuke-input"
      placeholder="www.lijinke"
      type="text"
      value=""
    />
    <span
      class="cuke-input-group-addon"
    >
      .cn
    </span>
  </span>
</div>
`;

exports[`<Input/> should render custom size 1`] = `
<div>
  <input
    class="cuke-input cuke-input-small"
    placeholder="small"
    type="text"
    value=""
  />
  <input
    class="cuke-input"
    placeholder="default"
    type="text"
    value=""
  />
  <input
    class="cuke-input cuke-input-large"
    placeholder="large"
    type="text"
    value=""
  />
</div>
`;

exports[`<Input/> should render input password 1`] = `
<div
  class="cuke-input-wrapper cuke-input-has-suffix"
>
  <input
    class="cuke-input cuke-input-password"
    placeholder="请输入"
    type="password"
    value="333"
  />
  <span
    class="cuke-input-suffix"
  >
    <div
      class="cuke-input-password-suffix"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
        />
        <circle
          cx="12"
          cy="12"
          r="3"
        />
      </svg>
    </div>
  </span>
</div>
`;

exports[`<Input/> should render prefix and suffix with input 1`] = `
<div>
  <div
    class="cuke-input-wrapper cuke-input-has-prefix"
  >
    <span
      class="cuke-input-prefix"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
    </span>
    <input
      class="cuke-input"
      placeholder="请输入"
      type="text"
      value=""
    />
  </div>
  <div
    class="cuke-input-wrapper cuke-input-has-suffix"
  >
    <input
      class="cuke-input"
      placeholder="请输入"
      style="margin:10px 0"
      type="text"
      value=""
    />
    <span
      class="cuke-input-suffix"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
    </span>
  </div>
  <div
    class="cuke-input-wrapper cuke-input-has-prefix cuke-input-has-suffix"
  >
    <span
      class="cuke-input-prefix"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
    </span>
    <input
      class="cuke-input"
      placeholder="请输入"
      type="text"
      value=""
    />
    <span
      class="cuke-input-suffix"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
    </span>
  </div>
</div>
`;
